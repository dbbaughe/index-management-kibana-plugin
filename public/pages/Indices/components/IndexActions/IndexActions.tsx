import React, { Component } from 'react';
import {
  EuiButton,
  EuiFlexGroup,
  EuiFlexItem,
} from '@elastic/eui';

interface IndexActionsProps {
  isAddDisabled: boolean;
  onClickAdd: Function;
}

export default class IndexActions extends Component<IndexActionsProps, object> {

  render() {
    const { isAddDisabled, onClickAdd } = this.props;
    return (
      <EuiFlexGroup justifyContent="spaceBetween" alignItems="center">
        <EuiFlexItem grow={false}>
          <EuiButton disabled={isAddDisabled} onClick={onClickAdd} data-test-subj="addPolicyButton">
            Add policy
          </EuiButton>
        </EuiFlexItem>
      </EuiFlexGroup>
    );
  }
}
