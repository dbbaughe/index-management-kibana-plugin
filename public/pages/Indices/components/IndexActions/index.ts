import IndexActions from './IndexActions';

export default IndexActions;
