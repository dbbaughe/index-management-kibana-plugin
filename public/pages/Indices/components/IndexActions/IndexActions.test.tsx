/*
 * Copyright 2019 Amazon.com, Inc. or its affiliates. All Rights Reserved.
 *
 * Licensed under the Apache License, Version 2.0 (the "License").
 * You may not use this file except in compliance with the License.
 * A copy of the License is located at
 *
 * http://www.apache.org/licenses/LICENSE-2.0
 *
 * or in the "license" file accompanying this file. This file is distributed
 * on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either
 * express or implied. See the License for the specific language governing
 * permissions and limitations under the License.
 */

import React from "react";
import "@testing-library/jest-dom/extend-expect";
import { render, fireEvent } from "@testing-library/react";
import IndexActions from "./IndexActions";

describe("<IndexActions /> spec", () => {
  it("renders the component", async () => {
    const { container } = render(<IndexActions isAddDisabled={false} onClickAdd={() => {}} />);

    expect(container.firstChild).toMatchSnapshot();
  });

  it("disables add button", async () => {
    const { getByTestId } = render(<IndexActions isAddDisabled={true} onClickAdd={() => {}} />);

    expect(getByTestId("indexActionsAddPolicy")).toBeDisabled();
  });

  it("calls add button", async () => {
    const onClickAdd = jest.fn();
    const { getByTestId } = render(<IndexActions isAddDisabled={false} onClickAdd={onClickAdd} />);

    expect(getByTestId("indexActionsAddPolicy")).toBeEnabled();
    fireEvent.click(getByTestId("indexActionsAddPolicy"));
    expect(onClickAdd).toHaveBeenCalled();
  });
});
