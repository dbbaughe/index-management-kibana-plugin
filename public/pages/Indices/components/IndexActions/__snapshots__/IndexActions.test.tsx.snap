// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<IndexActions /> spec renders the component 1`] = `
<div
  class="euiFlexGroup euiFlexGroup--gutterLarge euiFlexGroup--alignItemsCenter euiFlexGroup--justifyContentSpaceBetween euiFlexGroup--directionRow euiFlexGroup--responsive"
>
  <div
    class="euiFlexItem euiFlexItem--flexGrowZero"
  >
    <button
      class="euiButton euiButton--primary"
      data-test-subj="indexActionsAddPolicy"
      type="button"
    >
      <span
        class="euiButton__content"
      >
        <span
          class="euiButton__text"
        >
          Add policy
        </span>
      </span>
    </button>
  </div>
</div>
`;
